File: web-front-reactive/src/main/java/template/modules/admin/presentations/users/UserDto.java
package template.modules.admin.presentations.users;

import lombok.Data;
import lombok.NonNull;
import template.modules.admin.domain.users.User;

@Data
public class UserDto {
    private String name;
    private String login;
    private String email;
    private String avatar;

    public @NonNull User convertToModel(Long userId) {
        return User.create(userId, name, login, email, avatar);

    }
}

File: web-front-reactive2/src/main/kotlin/template/modules/admin/presentations/user/UserDto.kt
package template.modules.admin.presentations.user

import template.modules.admin.domain.users.User

data class UserDto(
    val name: String,
    val login: String,
    val email: String,
    val avatar: String? = null
)

fun UserDto.toModel(withId: Long? = null) = User(withId, this.name, this.login, this.email, this.avatar)
